src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -229,15 +229,21 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude) || mode == Mode.AppendVersion)
+            if (!string.IsNullOrEmpty(Href) && AppendVersion == true)
             {
-                // No globbing to do, just build a <link /> tag to match the original one in the source file.
-                // Or just add file version to the link tag.
-                BuildLinkTag(attributes, builder);
+                EnsureFileVersionProvider();
+                output.Attributes[HrefAttributeName].Value = _fileVersionProvider.AddFileVersionToPath(Href);
```

Comment:
Do not use `Href` here. That causes this helper to ignore changes earlier tag helpers may have made. Must instead use `output.Attributes[HrefAttributeName].Value as string`. Check that expression for `null` (not null or empty) rather than checking `string.IsNullOrEmpty(Href)` at all.